TMXLoader/TMXMap.cpp

```diff
@@ -14,165 +14,96 @@
-{
-	return m_version;
-}
-
-
-std::string TMXMap::getOrientation()
-{
-	return m_orientation;
-}
-
-
-unsigned int TMXMap::getWidth()
-{
-	return m_width;
-}
-
-
-unsigned int TMXMap::getHeight()
-{
-	return m_height;
-}
+    m_width = std::stoi(mapData[3]);
+    m_height = std::stoi(mapData[4]);
+    m_tileWidth = std::stoi(mapData[5]);
+    m_tileHeight = std::stoi(mapData[6]);
 
+    m_backgroundColour[0] = std::stoi(mapData[8]);
+    m_backgroundColour[1] = std::stoi(mapData[8]);
+    m_backgroundColour[2] = std::stoi(mapData[8]);
 
-unsigned int TMXMap::getTileWidth()
-{
-	return m_tileWidth;
+    m_propertiesMap = mapProps;
 }
 
+float TMXMap::getVersion() const noexcept { return m_version; }
 
-unsigned int TMXMap::getTileHeight()
-{
-	return m_tileHeight;
-}
+unsigned TMXMap::getWidth() const noexcept { return m_width; }
+unsigned TMXMap::getHeight() const noexcept { return m_height; }
+unsigned TMXMap::getTileWidth() const noexcept { return m_tileWidth; }
+unsigned TMXMap::getTileHeight() const noexcept { return m_tileHeight; }
 
+std::array<unsigned, 3> TMXMap::getBackgroundColor() const noexcept { return m_backgroundColour; }
 
-std::array<unsigned int, 3> TMXMap::getBackgroundColourArray()
-{
-    return m_backgroundColourArray;
-}
+std::string TMXMap::getOrientation() const noexcept { return m_orientation; }
+std::string TMXMap::getRenderOrder() const noexcept { return m_renderOrder; }
 
+void TMXMap::addTileSet(TMXTileSet const &newTileSet) noexcept { m_tileSetVector.push_back(newTileSet); }
 
-std::string TMXMap::getRenderOrder()
+TMXTileSet *TMXMap::getTileSet(std::string const &tileSetName) noexcept
 {
-    return m_renderOrder;
-}
-
-
-void TMXMap::addTileSet(TMXTileSet newTileSet)
-{
-	m_tileSetVector.push_back(newTileSet);
-}
-
-
-TMXTileSet* TMXMap::getTileSet(std::string tileSetName)
-{
-    for (unsigned int index = 0; index < m_tileSetVector.size(); ++index)
-    {
-        if (m_tileSetVector[index].getName() == tileSetName)
-        {
-            return &m_tileSetVector[index];
-        }
-    }
-    
+    for (unsigned idx{0}; idx < m_tileSetVector.size(); ++idx)
+        if (m_tileSetVector[idx].getName() == tileSetName)
+            return &m_tileSetVector[idx];
     std::cout << "TMXLoader: tileset layer '" << tileSetName << "' could not be found." << std::endl;
-    
     return nullptr;
 }
 
+void TMXMap::addLayer(TMXTileLayer const &newLayer) noexcept { m_layerVector.push_back(newLayer); }
 
-void TMXMap::addLayer(TMXTileLayer newLayer)
+TMXTileLayer *TMXMap::getTileLayer(std::string const &layerName) noexcept
 {
-	m_layerVector.push_back(newLayer);
-}
-
-
-TMXTileLayer* TMXMap::getTileLayer(std::string layerName)
-{
-    for (unsigned int index = 0; index < m_layerVector.size(); ++index)
-    {
-        if (m_layerVector[index].getName() == layerName)
-        {
-            return &m_layerVector[index];
-        }
-    }
-    
+    for (unsigned idx{0}; idx < m_layerVector.size(); ++idx)
```

Comment:
Since we're using modern C++, I'd suggest to use range-based for as well:
```c++
for (const auto &layer : m_layerVector)
```

Reply:
That is another good change. It's just that I'm not used to range-based for yet.